### PR TITLE
Pass product 'platforms' and 'cpe_items' directory to `ssg.template.Builder()` when loading test scenarios in Automatus

### LINKED ISSUE #9752: Automatus CI `TypeError: __init__() missing 2 required positional arguments: 'platforms_dir' and 'cpe_items_dir'`
#### Description of problem:
Automatus in CI traceback:
```
Traceback (most recent call last):
  File "/home/runner/work/content/content/tests/automatus.py", line 502, in <module>
    main()
  File "/home/runner/work/content/content/tests/automatus.py", line 498, in main
    options.func(options)
  File "/home/runner/work/content/content/tests/ssg_test_suite/rule.py", line 669, in perform_rule_check
    checker.test_target()
  File "/home/runner/work/content/content/tests/ssg_test_suite/oscap.py", line 699, in test_target
    self._test_target()
  File "/home/runner/work/content/content/tests/ssg_test_suite/rule.py", line 449, in _test_target
    test_content_by_rule_id = self._get_test_content_by_rule_id(
  File "/home/runner/work/content/content/tests/ssg_test_suite/rule.py", line 435, in _get_test_content_by_rule_id
    rule_test_content = self._get_rule_test_content(rule)
  File "/home/runner/work/content/content/tests/ssg_test_suite/rule.py", line 417, in _get_rule_test_content
    all_tests = self._load_all_tests(rule)
  File "/home/runner/work/content/content/tests/ssg_test_suite/rule.py", line 394, in _load_all_tests
    template_builder = ssg.templates.Builder(
TypeError: __init__() missing 2 required positional arguments: 'platforms_dir' and 'cpe_items_dir'
Error: Process completed with exit code 1.
```
https://github.com/ComplianceAsCode/content/actions/runs/3372700322/jobs/5596508651

#### SCAP Security Guide Version:
master

#### Steps to Reproduce:
So far I was not able to reproduce it locally.

#### Actual Results:
Traceback

#### Expected Results:
No traceback

#### Additional Information/Debugging Steps:
Problematic commit 422df0a4316

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Pass the product specific `platforms` and `cpe_items` directories to `ssg.templates.Builder()`.

#### Rationale:

- The Builder class requires the `platforms` and `cpe_items` directories to list all test scenarios.

- Fixes #9752

#### Review Hints:

-  Try to test a rule in a container or VM with and without this patch.
- Related:
  - https://github.com/ComplianceAsCode/content/actions/runs/3372700322/jobs/5596508651
  - https://github.com/ComplianceAsCode/content/actions/runs/3370997294/jobs/5592710001

### CODE REVIEW

**Review by mildas — APPROVED** [review]
Works for me :+1: 

@vojtapolasek merge if everything's ok from your point of view.

File: tests/ssg_test_suite/rule.py [hunk 1]
```diff
@@ -391,8 +391,10 @@ def _load_all_tests(self, rule):
         product_yaml = common.get_product_context(self.test_env.product)
         # Initialize a mock template_builder.
         empty = "/ssgts/empty/placeholder"
+        product_build_dir = os.path.join(common.SSG_ROOT, "build", product_yaml["product"])
         template_builder = ssg.templates.Builder(
-            product_yaml, empty, common._SHARED_TEMPLATES, empty, empty)
+            product_yaml, empty, common._SHARED_TEMPLATES, empty, empty,
+            product_build_dir + "/platforms", product_build_dir + "/cpe_items")
```
Comment (yuumasato) [inline tests/ssg_test_suite/rule.py:397]: We probably should have a constant in `ssg.common` defining `platforms` and `cpe_items`

**Comment by vojtapolasek** [comment]
Works for me as well. Yes, the direcotry names as strings was the only thing which caught my eye.

**Comment by yuumasato** [comment]
@vojtapolasek just pushed a commit moving the strings to constants. :)

**Comment by vojtapolasek** [comment]
@yuumasato  if we would like to be extra nitpicky, then you still have there the "build" directory defined as a string.

**Comment by jan-cerny** [comment]
yes, but it's hard to say if its extra or little

**Comment by yuumasato** [comment]
> @yuumasato if we would like to be extra nitpicky, then you still have there the "build" directory defined as a string.

It is not trivial to include "build" into the constant, as the `platforms` and `cpe_items` the `Builder()` is consuming are product specific.

I could add define the constant as `build/{}/platforms` and then use it like `_PLATFORMS_DIR.format(product_yaml["product"])`.

> but it's hard to say if its extra or little

@jan-cerny I'm not sure I understand,  little what?

**Comment by yuumasato** [comment]
Also, not sure if intentional or not, but before #9700, existence of `build/${PRODUCT}` was not required for Automatus to run.

**Comment by jan-cerny** [comment]
> I'm not sure I understand, little what?
little nitpicky

**Comment by yuumasato** [comment]
> @yuumasato if we would like to be extra nitpicky, then you still have there the "build" directory defined as a string.

Actually, to completly get rid of "build" string we need to add a new parameter to `Automatus` where we specify the build dir.

**Comment by vojtapolasek** [comment]
U see. I think we can leave it as it is now. @mildas  ?

**Comment by yuumasato** [comment]
@jan-cerny @vojtapolasek @mildas I have created an alternate solution #9755

> > @yuumasato if we would like to be extra nitpicky, then you still have there the "build" directory defined as a string.
> 
> Actually, to completly get rid of "build" string we need to add a new parameter to `Automatus` where we specify the build dir.

Instead of passing `platform-dirs` and `cpe-item-dirs` to Automatus, making it trickle down that info to `RuleChecker()` class, how about making `Builder()` work without those paths? Automatus doesn't need the data created by `Builder()` using the paths

**Comment by mildas** [comment]
> U see. I think we can leave it as it is now. @mildas ?

Yes, this solution is fine for me.

> Instead of passing platform-dirs and cpe-item-dirs to Automatus, making it trickle down that info to RuleChecker() class, how about making Builder() work without those paths? Automatus doesn't need the data created by Builder() using the paths

Could it not cause possible unexpected build behavior? E.g Templates being without CPEs. Would it be caught somewhere else where we do need to have CPE?

File: tests/ssg_test_suite/rule.py [hunk 2]
```diff
@@ -392,7 +392,10 @@ def _load_all_tests(self, rule):
         # Initialize a mock template_builder.
         empty = "/ssgts/empty/placeholder"
         template_builder = ssg.templates.Builder(
-            product_yaml, empty, common._SHARED_TEMPLATES, empty, empty)
+            product_yaml, empty, common._SHARED_TEMPLATES, empty, empty,
+            common.DEFAULT_BUILD_DIR + common.PROD_PLATFORMS_DIR.format(product_yaml["product"]),
+            common.DEFAULT_BUILD_DIR + common.PROD_CPE_ITEMS_DIR.format(product_yaml["product"]))
+
```
Comment (mildas) [inline tests/ssg_test_suite/rule.py:398]: Code climate doesn't like this line

**Comment by yuumasato** [comment]
I pretty much prefer the solution proposed in #9755, so I'm closing this PR.

**Pull request closed without merging**